integration-tests/testsetups/ocr.go

```diff
@@ -824,70 +823,106 @@ func (o *OCRSoakTest) setFilterQuery() {
+					continue
+				}
+
+				// Process the fetched logs
+				for _, event := range logs {
+					if o.OCRVersion == "1" {
+						answerUpdated, err := o.ocrV1Instances[0].ParseEventAnswerUpdated(event)
+						if err != nil {
+							o.log.Warn().
+								Err(err).
+								Str("Address", event.Address.Hex()).
+								Uint64("Block Number", event.BlockNumber).
+								Msg("Error parsing event as AnswerUpdated")
+							continue
+						}
+						o.log.Info().
 							Str("Address", event.Address.Hex()).
 							Uint64("Block Number", event.BlockNumber).
-							Msg("Error parsing event as AnswerUpdated")
-						continue
-					}
-					o.log.Info().
-						Str("Address", event.Address.Hex()).
-						Uint64("Block Number", event.BlockNumber).
-						Uint64("Round ID", answerUpdated.RoundId.Uint64()).
-						Int64("Answer", answerUpdated.Current.Int64()).
-						Msg("Answer Updated Event")
-				} else if o.OCRVersion == "2" {
-					answerUpdated, err := o.ocrV2Instances[0].ParseEventAnswerUpdated(event)
-					if err != nil {
-						o.log.Warn().
-							Err(err).
+							Uint64("Round ID", answerUpdated.RoundId.Uint64()).
+							Int64("Answer", answerUpdated.Current.Int64()).
+							Msg("Answer Updated Event")
+					} else if o.OCRVersion == "2" {
+						answerUpdated, err := o.ocrV2Instances[0].ParseEventAnswerUpdated(event)
+						if err != nil {
+							o.log.Warn().
+								Err(err).
+								Str("Address", event.Address.Hex()).
+								Uint64("Block Number", event.BlockNumber).
+								Msg("Error parsing event as AnswerUpdated")
+							continue
+						}
+						o.log.Info().
 							Str("Address", event.Address.Hex()).
 							Uint64("Block Number", event.BlockNumber).
-							Msg("Error parsing event as AnswerUpdated")
-						continue
-					}
-					o.log.Info().
-						Str("Address", event.Address.Hex()).
-						Uint64("Block Number", event.BlockNumber).
-						Uint64("Round ID", answerUpdated.RoundId.Uint64()).
-						Int64("Answer", answerUpdated.Current.Int64()).
-						Msg("Answer Updated Event")
-				}
-			case err = <-eventSub.Err():
-				backoff := time.Second
-				for err != nil {
-					o.log.Info().
-						Err(err).
-						Str("Backoff", backoff.String()).
-						Interface("Query", o.filterQuery).
-						Msg("Error while subscribed to OCR Logs. Resubscribing")
-					ctx, cancel = context.WithTimeout(testcontext.Get(o.t), backoff)
-					eventSub, err = o.seth.Client.SubscribeFilterLogs(ctx, o.filterQuery, eventLogs)
-					cancel()
-					if err != nil {
-						time.Sleep(backoff)
-						backoff = time.Duration(math.Min(float64(backoff)*2, float64(30*time.Second)))
+							Uint64("Round ID", answerUpdated.RoundId.Uint64()).
+							Int64("Answer", answerUpdated.Current.Int64()).
+							Msg("Answer Updated Event")
 					}
 				}
+
+				// Track last checked block number and update starting block number
+				lastCheckedBlockNum = latestBlock
+				startingBlockNum = latestBlock + 1
```

Comment:
Minor nit: can we just repurpose startingBlockNum to check for the lastCheckedBlockNum. Not a big deal but good for readability

Reply:
Thanks for the suggestion. Wanted to have a clear separation between the initial run and subsequent runs but it does not add that much value. Simpler code is always better.